frontend/admin/src/js/components/apiManagedTable/useTableState.ts

```diff
@@ -0,0 +1,210 @@
+import { useMemo } from "react";
+import { useSearchParams } from "react-router-dom";
+import isEqual from "lodash/isEqual";
+
+import { SearchState, SortingRule } from "./basicTableHelpers";
+
+export interface CommonTableState<T> {
+  currentPage?: number;
+  pageSize?: number;
+  hiddenColumnIds?: Array<string>;
+  sortBy?: SortingRule<T>;
+  searchState?: SearchState;
+}
+
+interface TableState<T, F> extends CommonTableState<T> {
+  filters: F;
+}
+
+type DefaultState<T, F> = Partial<TableState<T, F>>;
+type SetTableState<T, F> = (newState: TableState<T, F>) => void;
+
+const setInt = (
+  key: string,
+  params: URLSearchParams,
+  newValue?: number,
+  defaultValue?: number,
+) => {
+  const prevValue = params.get(key);
+
+  if (!prevValue || parseInt(prevValue, 10) !== newValue) {
+    if (newValue && newValue !== defaultValue) {
+      params.set(key, newValue.toString());
+    } else {
+      params.delete(key);
+    }
+  }
+
+  return params;
+};
+
+const setArray = <T>(
+  key: string,
+  params: URLSearchParams,
+  newValue?: Array<T>,
+  defaultValue?: Array<T>,
+) => {
+  const prevValue = params.get(key);
+  if (!prevValue || !isEqual(prevValue.split(","), newValue)) {
+    if (newValue && newValue !== defaultValue) {
```

Comment:
This should also use deep comparison, no?
```suggestion
    if (newValue && !isEqual(newValue, defaultValue)) {
```

Reply:
Yes! Thanks for catching that 😄